app/api/mail.py

```diff
@@ -0,0 +1,128 @@
+from app import app
+from flask import Blueprint, request, jsonify
+from app.api.api_helper import required_param, optional_param
+from app.util.authentication import auth_token_required, ERROR_LEVEL_API
+from app.model.email import Email, EmailStatus
+from app.util.db import db
+from app.util.mailer import mailer
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from sqlalchemy import desc, exc
+import bleach
+import json
+
+mail_api = Blueprint("mail_api", __name__, url_prefix='/api/mail')
+
+
+@mail_api.route('/send', methods=['POST'])
+@auth_token_required
+def sendmail():
+    """Send email to specific addresses.
+
+    .. :quickref: Send email
+
+    :param json (to, from, cc, bcc, subject, text):
+    :returns: int status_code, int email_id, str error_message
+    | Status code list:
+    | 200 - ok, message sent
+    | 500 - something goes wrong
+    :rtype: json
+    """
+
+    response_dictionary = {'status_code': 200}
+    response = {}
+    content = request.get_json(force=True)
+    mail_from = required_param(content, "from", str)
+    mail_to = required_param(content, "to", str)
+    mail_cc = optional_param(content, "cc", str)
+    mail_bcc = optional_param(content, "bcc", str)
+    mail_subj = bleach.clean(optional_param(content, "subject", str))
+    mail_body = bleach.clean(optional_param(content, "text", str))
+    mailto = [mail_to]
+    if mail_cc is not None:
+        mailto += [mail_cc]
+    else:
+        mail_cc = ""
+    if mail_bcc is not None:
+        mailto += [mail_bcc]
+    else:
+        mail_bcc = ""
+
+    # Prepare message
+    message = MIMEMultipart('alternative')
+    message["Subject"] = mail_subj
+    message["From"] = mail_from
+    message["To"] = mail_to
+    message["Cc"] = mail_cc
+    message["Bcc"] = mail_bcc
+    message.attach(MIMEText(mail_body, 'html'))
```

Comment:
why it's "html" and not "text"? the parameter name is "text", right? if you plan to provide just one mime part why to wrap it in multipart alternative? multipart alternative is usually used when you provide both text and html, text being an alternative to html in case email client can't show or doesn't want to show html.

Reply:
fixed